api/src/client/ApiTaxFilingClient.test.ts

```diff
@@ -0,0 +1,237 @@
+          { Name: "taxpayer-location", Value: randomInt(3).toString() },
+          { Name: "tax-business-name", Value: `Some, Name-${randomInt()}` },
+          { Name: "tax-name-control", Value: "Some" },
+        ]
+      : null,
+  Errors:
+    state != "SUCCESS"
+      ? [
+          {
+            Error:
+              state == "PENDING"
+                ? "There is no specific tax eligibility for this business at the moment but please do come back to check later since we get weekly updates."
+                : "No matching record found for the business. Please verify your Taxpayer ID and Business Name.",
+            Field: "Results",
+          },
+        ]
+      : null,
+  Results: state == "SUCCESS" ? generateTaxFilingResults() : null,
+  ...overrides,
+});
+
+const generateApiTaxFilingOnboardingResponse = (
+  overrides: Partial<ApiTaxFilingOnboardingResponse>,
+  statusCode: 200 | 400 | 500 | number = 200
+): ApiTaxFilingOnboardingResponse => {
+  return {
+    ApiKey: `some-ApiKey-${randomInt()}`,
+    Errors:
+      statusCode == 400
+        ? [
+            {
+              Error: `some-error-content-${randomInt()}`,
+              Field: randomInt() % 2 === 0 ? "Taxpayer ID" : "Business Name",
+            },
+          ]
+        : [],
+    Notice: statusCode == 200 ? "You are signed up for the Tax Calendar Reminder Service." : null,
+    StatusCode: statusCode as 200 | 400 | 500,
+    ...overrides,
+  };
+};
+
+describe("ApiTaxFilingClient", () => {
+  let client: ApiTaxFilingClient;
+  let logger: LogWriterType;
+  let taxIdAndBusinessNameAndEmail: { businessName: string; taxId: string; email: string };
+  const config = { apiKey: "abcdef", baseUrl: "example.com/gov2go" };
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    taxIdAndBusinessNameAndEmail = generateTaxIdAndBusinessNameAndEmail({});
+    logger = LogWriter("NavigatorWebService", "ApiLogs", "us-test-1");
+    client = apiTaxFilingClient(config, logger);
+  });
+
+  describe("lookup", () => {
+    it("posts to the endpoint with the api tax-filing lookup request object", async () => {
+      const stubResponse = generateApiTaxFilingLookupResponse({});
+      mockAxios.post.mockResolvedValue({ data: stubResponse });
+
+      const postBody: ApiTaxFilingLookupRequest = {
+        ApiKey: config.apiKey,
+        ServiceName: "Search",
+        Data: [
+          {
+            Name: "taxpayer-id",
+            Value: taxIdAndBusinessNameAndEmail.taxId,
+          },
+          {
+            Name: "tax-business-name",
+            Value: taxIdAndBusinessNameAndEmail.businessName,
+          },
+        ],
+      };
+
+      await client.lookup(taxIdAndBusinessNameAndEmail.taxId, taxIdAndBusinessNameAndEmail.businessName);
+      expect(mockAxios.post).toHaveBeenCalledWith(`${config.baseUrl}/lookup`, postBody);
+    });
+
+    it("returns taxFiling data on success", async () => {
+      const Results = generateTaxFilingResults(["test-id"], 1) ?? [];
```

Comment:
I don't understand this line. Is there a chance that your generator will return undefined? Why do you have the `?? []` here?

Reply:
In general, I prefer using overrides in the generators over passing arguments like this, so that the generators are dumb. When they start to have all this logic in them then it starts to get to a place where you might want to test your test helpers, which is a potentially long recursive loop haha. 

So I'd ask that:
- Why pass a number and list of IDs here instead of using overrides to set the sub-properties of this response like we do when using other generators?
- If you do think this is the best way to do it, at the very least, I think these things would be much more readable if they were nested in an object. Having a magic number (or boolean, or string) in the parameters of a function doesn't give you any insight into what it *means* - I have to into the definition of the function to understand why there is a `1` here and how it is even used. This could be really easily remedied by having the parameters of the generator be an object, so that you can label the parameters as you call the function. Like: `generateTaxFilingResults({ids: ["test-id"], numberOfX: 1 })`

** I want note here that I wrote `numberOfX` because I went into the definition of the generator and tried for 20 seconds to parse out what the number parameter was being used for. Number of dates? Number of values? The code there with all that logic is not very clear and that is sort of my point when it comes to adding complex logic to the generator functions, which are typically a more "dumb" helper